RecyclerView/src/main/kotlin/com/infomaniak/core/recyclerview/ResettableLazy.kt

```diff
@@ -30,7 +30,8 @@ internal class ResettableLazy<T : Any>(private val initializer: () -> T) {
         return value ?: initializer().apply { value = this }
     }
 
-    fun invalidate() {
-        value = null
-    }
+    /**
+     * Returns the value prior to invalidation/removal.
+     */
+    fun invalidate(): T? = value.also { value = null }
```

Comment:
Is it not strange that the invalidate() method suddenly returns something?

Could we make `value` public with `private set` for situations where someone needs finer control because you actually need to know if the value you're dealing with is null or not?

Reply:
Agreed, it makes more sense to expose `value` with private setter, done.